addon/components/ember-popper-base.js

```diff
@@ -0,0 +1,202 @@
+   */
+  @property _updateRAF = null
+
+  // ================== LIFECYCLE HOOKS ==================
+
+  didUpdateAttrs() {
+    this._popper.destroy();
+
+    this._updateRAF = requestAnimationFrame(() => {
+      const target = this.get('_popperTarget');
+      const element = this.get('_popperElement');
+      const options = this.get('_options');
+
+      this._addPopper(target, element, options);
+    });
+  }
+
+  didInsertElement() {
+    super.didInsertElement(...arguments);
+
+    const target = this.get('_popperTarget');
+    const element = this.get('_popperElement');
+    const options = this.get('_options');
+
+    this._addPopper(target, element, options);
+  }
+
+  willDestroyElement() {
+    super.willDestroyElement(...arguments);
+
+    this._popper.destroy();
+    cancelAnimationFrame(this._updateRAF);
+  }
+
+  /**
+   * ================== ACTIONS ==================
+   */
+
+  @action
+  update() {
+    this._popper.update();
+  }
+
+  @action
+  enableEventListeners() {
+    this._popper.enableEventListeners();
+  }
+
+  @action
+  disableEventListeners() {
+    this._popper.disableEventListeners();
+  }
+
+  // ================== PRIVATE IMPLEMENTATION DETAILS ==================
+
+  _addPopper(target, element, options) {
+    this._popper = new Popper(target, element, options);
+  }
+
+  /**
+   * The element to be positioned by the Popper.
+   */
+  @computed('_renderInPlace')
+  _popperElement() {
+    assert('_popperElement must only be used after the component is rendered', !!this.element);
+
+    const renderInPlace = this.get('_renderInPlace');
+
+    // If not rendering in place, return the wrapper around yielded elements. See template for details.
+    return renderInPlace ? this.element : document.querySelector(`.popper-${this.elementId}`);
+  }
+
+  /**
+   * Used to manually set _popperTarget in `didInsertElement()`. Need to wait until
+   * `didInsertElement()` to avoid calling `document.querySelectorAll(this.get('target'))`
+   * before the target exists.
+   */
+  @computed('target')
+  _popperTarget() {
+    const target = this.get('target');
+
+    let popperTarget;
+
+    // If there is no target, set the target to the parent element
+    if (!target) {
+      popperTarget = this.element.parentNode;
+    } else if (target instanceof Element) {
+      popperTarget = target;
+    } else {
+      const nodes = document.querySelectorAll(target);
+
+      assert(`ember-popper with target selector "${target}" found ${nodes.length}`
+             + 'possible targets when there should be exactly 1', nodes.length === 1);
+
+      popperTarget = nodes[0];
+    }
+
+    return popperTarget;
+  }
+
+  @computed('popperContainer', '_renderInPlace')
+  _popperContainer() {
+    const renderInPlace = this.get('_renderInPlace');
+    const maybeContainer = this.get('popperContainer');
+
+    let popperContainer;
+
+    if (renderInPlace) {
+      popperContainer = this.element.parentNode;
```

Comment:
This borks fastboot since it doesn't have access to `this.element`. Previously we did [this](https://github.com/kybishop/ember-popper/blob/master/addon/components/ember-popper.js#L45).

As long as we have a similar check for `self.document` and bail early, we should be good.